README.md

```diff
@@ -1,19 +1,84 @@
 # NestJS Throttler Package
 
-An Express request throttler for controllers and routes.
+A Rate-Limiter for NestJS, regardless of the context.
 
-## Decorator
+This package comes with a couple of goodies that should be mentioned, first is the `ThrottlerModule`.
 
+## ThrottlerModule
+
+The `ThrottleModule` is the main entry point for this package, and can be used in a synchronous or asynchronous manner. All the needs to be passed is the `ttl`, the time to live in seconds for the request tracker, and the `limit`, or how many times an endpoint can be hit before returning a 429.
+
+```ts
+@Module({
+  imports: [
+    ThrottlerModule.forRoot({
+      ttl: 60,
+      limit: 10,
+    }),
+  ],
+})
+export class AppModule {}
+```
+
+The above would mean that 10 requests from the same IP can be made to a single endpoint in 1 minute.
+
+```ts
+@Module({
+  imports: [
+    ThrottlerModule.forRootAsync({
+      imports: [ConfigModule],
+      inject: [ConfigService],
+      useFactory: (config: ConfigService) => ({
+        ttl: config.get('RATE_LIMIT_TTL'),
+        limit: config.get('RATE_LIMIT_LIMIT'),
```

Comment:
Could we change `RATE_LIMIT_` to `THROTTLER_` as a prefix so people use a more package-named prefix? Personal pref 🙃

Reply:
Yeah, no problem.